mccode/src/instrument.y

```diff
@@ -306,25 +307,25 @@ comp_share: /* empty */
       {
         $$ = codeblock_new();
       }
-    | "SHARE" codeblock comp_share_copy_extend
+    | "SHARE" codeblock comp_share_inherit_extend
       {
         struct code_block *cb;
         cb = codeblock_new();
         list_cat(cb->lines, $2->lines);
         list_cat(cb->lines, $3->lines);
         $$ = cb;
       }
-    | "SHARE" comp_share_copy_extend
+    | "SHARE" comp_share_inherit_extend
       {
         $$ = $2;
       }
 ;
 
-comp_share_copy_extend: /* empty */
+comp_share_inherit_extend: /* empty */
```

Comment:
This matches
- `SHARE`
- `SHARE %{...%}`
- `SHARE EXTEND %{...%}`
making the `EXTEND` keyword superfluous. 

Explicitly making the keyword optional (by ignoring its token, in the same way that comments are ignored) would simplify this repeated set of rules.